### ISSUE TITLE
Android: Seems broken in rn ^0.15.0

### ISSUE THREAD

**Issue by jforaker** [brentvatne/react-native-linear-gradient#43]
[Android] Ever since I upgraded to react-native `^0.15.0`, I no longer see the gradient. No errors in the android/chrome console.

Using `1.1.0-alpha`

Was working nicely before I updated. Any ideas?

**Comment by nickhudkins** [comment]
Just closed #40 which I was unable to reproduce on 0.15.0-rc with 1.1.0-alpha, can you make sure that no files ended up modified in the upgrade that could have undone some setup?

What flavor of Android are you running, I can try to give it a shot here.

**Comment by jforaker** [comment]
Android 6.0 / API 23 on a Nexus 5x

**Comment by nickhudkins** [comment]
Have not tested with Android 6.0 I'll give it a go now :+1:

**Comment by briansturgeon** [comment]
recommend checking the motherboard

**Comment by nickhudkins** [comment]
6.0 / API 23, Nexus 5x I was able to get running here: 

<img width="453" alt="screen shot 2015-11-25 at 6 04 18 pm" src="https://cloud.githubusercontent.com/assets/299126/11411560/1e30272a-939f-11e5-887b-2e5dda9d5df9.png">

Can you make sure no build.gradle files or anything of the sort ended up modified?

**Comment by jforaker** [comment]
Interesting. When I try to run that specific emulator in Genymotion, it completely crashes my Macbook pro, with some black screen with chinese(?) writing. Wow never seen that before, and never want to again.

Thanks for checking. I'm at a loss for now. Not a showstopper, but was a definite wow-factor when the client saw it :/

**Comment by nickhudkins** [comment]
If you have an example of it broken I can try to give it a shot over here.

**Comment by jforaker** [comment]
How do you mean? Its a large app with lots going on (api/data etc), so I cant really post an rn-play gist.

**Comment by nickhudkins** [comment]
Wasn't sure if it was something you could share and rip down to a
simplified test case (but keeping your dependencies etc).

**Comment by dgladkov** [comment]
The problem is that `LinearGradientManager` relies on `ViewManager.updateView` [android/src/main/java/com/BV/LinearGradient/LinearGradientManager.java#L62-L73](https://github.com/brentvatne/react-native-linear-gradient/blob/493f3fbcbe414f7a922913b42f506207da6fa117/android/src/main/java/com/BV/LinearGradient/LinearGradientManager.java#L62-L73) that was deprecated for some time and removed in https://github.com/facebook/react-native/commit/05c9a1f91aa27bb7a15054868f0189554db0c214. Unfortunately, I could not find a quick solution for this problem, probably needs refactoring of `LinearGradientView`.

**Comment by nickhudkins** [comment]
Well awesome to hear what's going on. Strange that I could not reproduce
but I'll take a look and get it refactored unless you'd like to take a stab
at it :)

**Comment by gadiel** [comment]
any update on this?

**Comment by nickhudkins** [comment]
I have yet to be able to reproduce, but if someone can provide a broken project I would be happy to look further into it. I can also take a look later this evening to see if I can finally reproduce it.

**Comment by brentvatne** [comment]
@nickhudkins - seems like this: https://github.com/facebook/react-native/issues/4154

**Comment by mattds** [comment]
Also getting this issue, I'm using RN 0.16.0 though. Got it working temporarily whilst a fix is in development, so thought I would share a temporary workaround. I removed updateView from LinearGradientManager and replaced

```javascript
protected FrameLayout createViewInstance(ThemedReactContext context) {
        return new FrameLayout(context);
    }
```

with

```javascript
    @Override
    protected FrameLayout createViewInstance(ThemedReactContext context) {
        WritableMap defaults = new WritableNativeMap();
        WritableArray array = new WritableNativeArray();
        array.pushInt(0);
        array.pushInt(0);

        defaults.putArray("colors", array);

        mGradientView = new LinearGradientView(context, new CatalystStylesDiffMap(defaults));

        FrameLayout frame = new FrameLayout(context);

        frame.removeAllViews();
        frame.addView(mGradientView);

        return frame;
    }
```

Also used the imports:
```javascript
import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.bridge.WritableNativeArray;
import com.facebook.react.bridge.WritableNativeMap;
```

**Comment by brentvatne** [comment]
Thanks @mattds! I tried that out and it worked great. Open to other issues to improve it, but for now I just shipped this as 1.3.0 so people can continue using. Also added Android to the example project :)

**Comment by mattds** [comment]
It's a pleasure :)
